Joss (#3)
poissonconsulting/bbousuite: paper/joss_preview/paper.Rmd

---
title: 'bbousuite: A set of R packages to facilitate analysis of Boreal Caribou survival and recruitment data'
authors:
- name: Seb Dalgarno
  orcid: 0000-0002-3658-4517
  affiliation: 1
- name: John Boulanger
  orcid: 0000-0001-8222-1445
  affiliation: 2
- name: Ayla Pearson
  orcid: 0000-0001-7388-1222
  corresponding: true
  affiliation: 1
- name: Joe Thorley
  orcid: 0000-0002-7683-4592
  affiliation: 1
- name: Troy Hegel  
  affiliation: 3
- name: Barry Nobert
  affiliation: 3
- name: Dave Hervieux
  affiliation: 3
affiliations:
- index: 1
  name: Poisson Consulting, Canada
- index: 2
  name: Integrated Ecological Research, Canada
- index: 3
  name: Province of Alberta, Canada
date: 6 May 2024
bibliography: paper.bib
output: 
  rticles::joss_article: default
csl: apa.csl
journal: JOSS
tags:
  - boreal caribou
  - Bayesian
  - maximum likelihood
  - survival
  - recruitment
  - population growth
---

# Summary

> Although many animal populations are managed based on their abundance [@williams_analysis_2002], reliable abundance estimates can be impractical to obtain for mobile herding populations with large ranges such as boreal caribou (*Rangifer tarandus caribou*).

Reviewer: ```suggestion
Although many animal populations are managed based on their abundance [@williams_analysis_2002], reliable abundance estimates can be difficult to obtain for mobile herding populations with large ranges such as boreal caribou (*Rangifer tarandus caribou*).
```

> Consequently, such populations are commonly managed based on their population growth rate ($\lambda$), which can be estimated from the population recruitment and adult female survival rates [@hatter_moose_1991]. 

Reviewer: ```suggestion
Consequently, such populations are commonly managed based on their population growth rate ($\lambda$), which can be estimated from the population recruitment ($R$) and adult female survival ($S$) rates [@hatter_moose_1991] using the following equation. 

$$\lambda = \frac{S}{1- R}$$
```

> The `bbou` suite is a set of R packages [@r_core_team_2023] to facilitate simulation and analysis of boreal caribou survival and recruitment data to estimate population growth.  

Reviewer: We need to update to R 2024 for the citation.

Reviewer 2: done

It consists of:  
- [`bboudata`](https://poissonconsulting.github.io/bboudata/), a collection of simulated and anonymized empirical survival and recruitment datasets.  
- [`bboutools`](https://poissonconsulting.github.io/bboutools/), tools to estimate population growth in a Bayesian or Maximum Likelihood (ML) framework, implemented under the hood using the `Nimble` R package [@de_valpine_programming_2017].  

> - [`bboushiny`](https://poissonconsulting.github.io/bboushiny/), a Graphical User Interface (GUI) to `bboutools` (\autoref{fig:bboushiny}).    

Reviewer: The autoref is not working when I render to html locally? 

<img width="954" alt="Screenshot 2024-09-05 at 07 14 27" src="https://github.com/user-attachments/assets/cfae9ddf-9496-468e-9fa3-617c995f9cba">

Reviewer 2: yes I'm not worried about this as it works when render as pdf and joss article

- [`bbouretro`](https://poissonconsulting.github.io/bbouretro/), tools to estimate population growth using traditional frequentist methods.  
- [`bbousims`](https://poissonconsulting.github.io/bbousims/), tools to simulate survival and recruitment data from hypothetical collaring and composition surveys.  

Each R package has a website with function documentation and a 'Get Started' guide.

> There are also several vignettes, including on `bboutools` [analytical methods](https://poissonconsulting.github.io/bboutools/articles/methods.html) and [prior selection](https://poissonconsulting.github.io/bboutools/articles/priors.html); `bbouretro` [analytical methods](https://poissonconsulting.github.io/bbouretro/articles/retro-methods.html); using [`bbousims` with `bboutools`](https://poissonconsulting.github.io/bbousims/articles/bboutools.html); and comparison of Bayesian and traditional method estimates with [empirical](https://poissonconsulting.github.io/bbousuite/articles/empirical-comparisons.html) and [simulated](https://poissonconsulting.github.io/bbousuite/articles/simulations.html) data. 

Reviewer: I think we need to add a to "and comparison of Bayesian" to get "and a comparison of Bayesian"

Reviewer 2: done

> ![bboushiny GUI. \label{fig:bboushiny}](figures/bboushiny-survival.png)

Reviewer: ```suggestion
![A screenshot of the bboushiny GUI. \label{fig:bboushiny}](figures/bboushiny-survival.png)
```

# Statement of need

Boreal caribou are found in most provinces and territories in Canada and have been listed as threatened since 2003 [@ECCC_2023].
Each jurisdiction has their own monitoring program responsible for boreal caribou.
Numerous methods have been used to estimate population growth, which can make cross-jurisdictional conversations on the health of the species complex. 

The suite of R packages in `bbousuite` aim to address this problem.

> `bboutools` provides a standardized methodology with simple, general models and sensible defaults for estimating survival, recruitment and population growth that can be compared across jurisdictions.

Reviewer: ```suggestion
`bboutools` provides a standardized methodology with simple, general models and reasonable defaults for estimating survival, recruitment and population growth values that can be compared across jurisdictions.
```

In addition, `bboushiny` provides access to `bboutools` functionality in a GUI; `bbouretro` facilitates comparison with traditional methods; and `bbousims` facilitates comparison of various methods' ability to recover known parameter values from simulated data. 

There is an existing web-based application for estimating boreal caribou population growth rate [@eacker_webbased_2019]. 

> However, users reported that it was difficult to use, especially if not familiar with Bayesian statistics, and had too many options to be considered a standarized method.  

Reviewer: ```suggestion
However, users reported that it had too many options to be considered a standarized method.  
```

The `caribouMetrics` R package, and associated shiny app `BayesianCaribouDemographicProjection`, also have functionality to estimate boreal caribou demographic parameters.  
However, the primary goal of these tools is to forecast boreal caribou population under different monitoring and disturbance scenarios [@dyson_2022].

# Overview of methods

Despite key methodological differences, which are explained in more detail below, `bboutools` and `bbouretro` use the same underlying equations to estimate survival, recruitment and population growth. 

> Survival is estimated from the monthly fate of collared adult females using the staggered entry Kaplan-Meier method [@pollock_survival_1989], with an option to include uncertain mortaliites in the total monthly mortalities summed prior to model fitting. 

Reviewer: ```suggestion
Survival is estimated from the monthly fate of collared adult females using the staggered entry Kaplan-Meier method [@pollock_survival_1989], with an option to include uncertain mortalites in the total monthly mortalities summed prior to model fitting. 
```

> Survival is estimated from the monthly fate of collared adult females using the staggered entry Kaplan-Meier method [@pollock_survival_1989], with an option to include uncertain mortaliites in the total monthly mortalities summed prior to model fitting. 

Reviewer: Does bboutools really use the Kaplan-Meier method?

Reviewer 2: removed suggestion that the underlying methods are shared

Recruitment is estimated from annual composition surveys following methods in DeCesare et al. [-@decesare_estimating_2012], where the adjusted recruitment accounts for recruitment of calves into the yearling/adult age class at the end of the caribou year.

> Population growth ($\lambda$) is estimated using the Hatter-Bergerud method [@hatter_moose_1991].

Reviewer: ```suggestion
Population growth ($\lambda$) is estimated using the Hatter-Bergerud equation [@hatter_moose_1991] presented above.
```

> Both `bboutools` and `bbouretro` allow the user to adjust the start month of the biological year and data are aggregated by biological year prior to model fitting.

Reviewer: I think the phrase aggregate is slightly misleading as the survival is still performed on monthly collar numbers.

Reviewer 2: good point - maybe something like:
"adjust the start month of the biological year and year for each observation is reassigned prior to model fitting."

A full comparison of `bboutools` and `bbouretro` statistical methods is beyond the scope of this paper.

> A key difference is that `bboutools` methods are parametric, whereas `bbouretro` methods are non-parametric. 

Reviewer: I'm also not sure that I agree that bbouretre is non-parametric. We should discuss.

Reviewer 2: ok sounds good - this section of text was taken from John's writeup - https://poissonconsulting.github.io/bbousuite/articles/empirical-comparisons.html#differences-and-similarities-in-traditional-and-bayesian-approaches

Reviewer 2: removed reference to parameter/non-parametric

Parametric methods assume the data (sample) are drawn from a population which can be described by a statistical distribution, i.e., the binomial in the case of survival and recruitment models. 

> Inference is based on the underlying statistical model, which allows for inference on statistical significance of model parameters. 

Reviewer: Do we focus on statistical significance in the software?

Reviewer 2: good question - it's based on John's text
we don't present statistical significance at any point

should rephrase to "which allows more precise estimation of uncertainty of model parameters"?

$$
\begin{aligned}
Calves_{j} & \sim \text{Binomial}(X_{j}, AdultFemales_{j}) \\
\text{logit}(\delta_{j}) & = \beta_{0} + \alpha_{j} + \beta_{j} \cdot Year_{j} \\
\alpha_{j} & \sim \text{Normal}(0, \sigma_{\alpha}) \\
FemaleYearlings_{j} & \sim \text{Binomial}(sex\_ratio, Yearlings_{j}) \\
Cows_{j} & \sim \text{Binomial}(adult\_sex\_ratio, CowsBulls_{j}) \\
AdultFemalesOther_{j} & \sim \text{Binomial}(adult\_sex\_ratio, AdultsUnknown_{j}) \\
AdultFemales_{j} & = FemaleYearlings_{j} + Cows_{j} + AdultFemalesOther_{j} \\
\end{aligned}
$$

where $X_{j}$ is the calves per adult female in the $j^{th}$ year, $\alpha_{j}$ is an annual random effect with SD $\sigma_{\alpha}$ and $\beta_{1}$ represents the trend. 

The model includes demographic stochasticity. 
Groups are aggregated by year prior to model fitting.
The sex ratio is fixed and can be adjusted by the user, with default of 0.5.
The adult female proportion can be estimated from counts of cows and bulls or fixed, with a default of 0.65, which accounts for higher mortality of males [@smith_2004].

We compared estimates from `bbouretro` and `bboutools` on anonymized empirical and simulated data.

> [Empirical comparisons](https://poissonconsulting.github.io/bbousuite/articles/empirical-comparisons.html) demonstrate that `bbouretro` methods yield similar estimates to Bayesian fixed-effects models with uninformative priors (\autoref{fig:empirical}).

Reviewer: As before autoref not working on my machine.

> By default, `bboutools` uses uninformative priors and these can be adjusted by the user. 

Reviewer: Are they truly uninformative?

Reviewer 2: Originally I had it as biologically reasonable vaguely informative priors
but im not too sure about this since:
1. we did some simple sensitivity analysis to ensure that priors weren't influencing results
2. the standard deviations are very large. the only informative prior I think is for adult_female_proportion

<img width="873" alt="image" src="https://github.com/user-attachments/assets/208e65f2-fbae-4c3c-9130-f7f0a1215766">

<img width="884" alt="image" src="https://github.com/user-attachments/assets/79b66027-7576-4af4-8454-62ae429fc6d7">

Reviewer: Depends who you ask - I think uninformative is reasonable based on all the work we did

> Estimates from random effects models tend to be less extreme than traditional and fixed effect models.

Reviewer: ```suggestion
Estimates from random effects models tend to be provide more reliable estimates than traditional and fixed effect models.
```

Reviewer 2: I'm unsure about this change. This paragraph is referring to the empirical comparison (not simulated data). My intention was to point out that random effect estimates are closer to the mean than the fixed effect estimates. However at this point I'm not sure the empirical comparison can show us that the random effect estimates are more reliable - that comes further down with simulations

Reviewer 2: although I agree "tend to be less extreme" doesn't sound great either...

Reviewer: We could go with "less noisy"?

Reviewer 2: yes sounds good

This is explored in more detail below.

> ![Comparison of traditional and Bayesian estimates of survival and recruitment by year.\label{fig:empirical}](figures/methods-compare.png)  

Reviewer: ```suggestion
![Comparison of traditional and Bayesian estimates of survival and recruitment by year. The bars represent 95% confidence/credible intervals.\label{fig:empirical}](figures/methods-compare.png)  
```

Data simulation (i.e., with `bbousims`) allows for comparison of various methods' ability to recover known parameter values (i.e., with bias, precision, coverage). 

[Our analysis](https://poissonconsulting.github.io/bbousuite/articles/simulations.html) demonstrates that the Bayesian random effects model generally performs best. 

> For example, a comparison of the distribution of bias in point estimates of population growth for 100 simulations and 20 years shows that the random effects model has the smallest spread in bias, especially at lower sample sizes (\autoref{fig:bias}).

Reviewer: ```suggestion
For example, a plot of the of bias in point estimates of population growth for 100 simulations and 20 years shows that the random effects model has the greatest precision, especially at lower sample sizes (\autoref{fig:bias}).
```

![Bias (% difference) in annual population growth point estimates and known population growth for 100 simulations and 20 years, by sample size and statistical method.\label{fig:bias}](figures/lambda-bias.png)

> Random effect models exhibit 'shrinkage', which causes more extreme estimates to be pulled towards the grand mean, especially when underlying data are sparse [@kery_bayesian_2011].

Reviewer: ```suggestion
Random effect models exhibit 'shrinkage', which causes more extreme estimates to be pulled towards the group mean, especially when underlying data are sparse [@kery_bayesian_2011].
```
